src/ad-services/identity-setup/index.ts

```diff
@@ -0,0 +1,24 @@
+import { communicationService, eventsRepository } from '@ad-engine/communication';
+import { BaseServiceSetup, utils } from '@ad-engine/core';
+
+export class IdentitySetup extends BaseServiceSetup {
+	private logGroup = 'identity-setup';
+	private identityReady: () => void;
+	private fallback = setTimeout(() => {
+		utils.logger(this.logGroup, 'Fallback launched');
+		this.identityReady();
+	}, 2500);
+
+	call(): Promise<void> {
+		const identityPromise = new Promise<void>((res) => {
+			this.identityReady = res;
+		});
+
+		communicationService.on(eventsRepository.IDENTITY_ENGINE_READY, () => {
+			this.identityReady();
+			clearTimeout(this.fallback);
+			utils.logger(this.logGroup, 'initialized');
+		});
+		return identityPromise;
```

Comment:
```suggestion
		return new Promise((resolve) => {
		    communicationService.on(eventsRepository.IDENTITY_ENGINE_READY, () => {
			resolve();
			clearTimeout(this.fallback);
			utils.logger(this.logGroup, 'initialized');
		    });
		});
```
it should work just the same, but you won't need that `identityReady` property which I guess will reduce bundle size a bit